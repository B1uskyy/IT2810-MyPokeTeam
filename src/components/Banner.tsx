import '../styles/banner.css';
import { useEffect, useState } from 'react';
import { useLocation } from 'react-router-dom';

interface BannerProps {
  title: string;
  text: string;
  desktopImage: string;
  mobileImage: string;
}

function Banner({ title, text, desktopImage, mobileImage }: BannerProps) {
  const [bgImage, setBgImage] = useState(desktopImage);
  const [isLoaded, setIsLoaded] = useState(false);
  const location = useLocation();

  useEffect(() => {
    const updateBackground = () => {
      if (window.innerWidth <= 768) {
        setBgImage(mobileImage);
      } else {
        setBgImage(desktopImage);
      }
    };

    updateBackground();
    window.addEventListener('resize', updateBackground);

    setIsLoaded(true);

    return () => {
      window.removeEventListener('resize', updateBackground);
    };
  }, [desktopImage, mobileImage, location.pathname]);

  return (
    <header className="banner" style={{ backgroundImage: `url(${bgImage})` }}>
      <div className={`banner-content ${isLoaded ? 'slide-in' : ''}`}>
        <h1>{title}</h1>
        <p>{text}</p>
      </div>
    </header>
  );
}

export default Banner;
