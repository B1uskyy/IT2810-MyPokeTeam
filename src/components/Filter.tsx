import React, { useEffect, useState } from 'react';
import '../styles/filter.css';

type FilterProps = {
  name: string;
  values: string[];
  onFilterChange: (selectedValue: string) => void;
};

export default function Filter({ name, values, onFilterChange }: FilterProps) {
  const [selectedValue, setSelectedValue] = useState('');

  const handleChange = (e: React.ChangeEvent<HTMLSelectElement>) => {
    const selected = e.target.value;
    setSelectedValue(selected);
    onFilterChange(selected);
    sessionStorage.setItem('filter', selected);
  };

  useEffect(() => {
    const savedFilter = sessionStorage.getItem('filter');
    if (savedFilter) {
      setSelectedValue(savedFilter);
    }
  }, [selectedValue]);
  return (
    <div className="pokemon-types">
      <label htmlFor={name.toLowerCase()}>Filter on type</label>
      <select id={name.toLowerCase()} value={selectedValue} onChange={handleChange} aria-label={`Select a ${name} `}>
        <option value="">All types</option>
        {values.map((value, index) => (
          <option key={index} value={value}>
            {value}
          </option>
        ))}
      </select>
    </div>
  );
}
