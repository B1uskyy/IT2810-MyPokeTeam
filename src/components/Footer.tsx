import '../styles/footer.css';
import pokeball from '../assets/pokeball.png';
export default function Footer() {

  return (
    <footer className="footer">
        <img className ="pokeball" src={pokeball} alt="pokeball" />
        <h3 className="info">Thank you for using team builder</h3>
    </footer>
  )
}