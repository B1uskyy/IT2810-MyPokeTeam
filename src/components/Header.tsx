import React from 'react';
import './Header.css';

const Header: React.FC = () => {
  return (
    <header className="header">
      <h1>Team-builder</h1>
      <p>
        Vivamus sit amet egestas magna. Aenean et lectus vitae massa interdum hendrerit sed vel turpis. Nulla consequat, augue ut condimentum vulputate.
      </p>
    </header>
  );
};

export default Header;