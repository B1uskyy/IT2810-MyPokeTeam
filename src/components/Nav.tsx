import pokeball from '../assets/pokeball.png';
import '../styles/nav.css';
import { useState } from 'react';
import { Link } from 'react-router-dom';

function Nav() {
  const [isOpen, setIsOpen] = useState(false);

  const toggleMenu = () => {
    setIsOpen(!isOpen);
  };

  const closeMenu = () => {
    setIsOpen(false);
  };

  return (
    <nav className="header">
      <div className="logo">
        <Link to="/home" onClick={closeMenu}>
          <img src={pokeball} alt="pokeball" />
        </Link>
      </div>
      <div className={`nav ${isOpen ? 'open' : ''}`}>
        <ul className="nav-links">
          <li>
            <Link className="nav-button" to="/home" onClick={closeMenu}>
              Explore Pok√©mons
            </Link>
          </li>
          <li>
            <Link className="nav-button" to="/team" onClick={closeMenu}>
              Your Team
            </Link>
          </li>
        </ul>
      </div>
      <div className={`hamburger ${isOpen ? 'open' : ''}`} onClick={toggleMenu}>
        <span className="bar"></span>
        <span className="bar"></span>
        <span className="bar"></span>
      </div>
    </nav>
  );
}

export default Nav;
