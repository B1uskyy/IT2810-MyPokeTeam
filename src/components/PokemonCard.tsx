import '../styles/pokemonCard.css';
import Cradily from '../assets/cradilypng.png';
import { FaRegStar, FaStar } from 'react-icons/fa';

export type Pokemon = {
  name: string;
  type: string;
  id: number;
  isFavorite: boolean;
  onToggleFavorite?: () => void;
};

export default function PokemonCard({ name, id, isFavorite, onToggleFavorite }: Pokemon) {
  let handleLeftclick;
  let handleRightclick;

  return (
    <section className="pokemonCard">
      <div className="pokemonName">
        <h2 className="pokemonNameHeader">{name}</h2>
      </div>
      <div className="icon">
        <img className="pokemonImage" src={Cradily} alt="" />
      </div>
      <div className="buttons">
        <button className="leftButton" onClick={handleLeftclick}>
          prevNum
        </button>
        <button className="rightButton" onClick={handleRightclick}>
          nextNum
        </button>
      </div>
      <button onClick={onToggleFavorite}>
        {!isFavorite && <FaRegStar size={30} data-testid="non-favorite-icon" />}
        {isFavorite && <FaStar size={30} color="yellow" data-testid="favorite-icon" />}
      </button>
      <div className="pokemonNumber">
        <h3>{id}</h3>
      </div>
      <div className="types">
        <div className="leftType"></div>
        <div className="rightType"></div>
      </div>
    </section>
  );
}
