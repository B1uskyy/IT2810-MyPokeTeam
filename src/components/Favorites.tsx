import '../styles/favorites.css';
import React, { useEffect, useState } from 'react';
import { RiPencilFill } from 'react-icons/ri';
import { IoIosSave } from 'react-icons/io';
import PokemonCard from './PokemonCard.tsx';
import { ListPokemon } from '../interfaces/pokemons.tsx';

interface FavoritesProps {
  pokemons: ListPokemon[];
  onToggleFavorite: (pokemon: ListPokemon) => void;
  isFavorited: (pokemon: ListPokemon) => boolean;
}

export function Favorites({ pokemons, onToggleFavorite, isFavorited }: FavoritesProps) {
  const [name, setName] = useState(() => {
    const savedName = localStorage.getItem('teamName');
    return savedName || 'Team name';
  });
  const [isEditing, setIsEditing] = useState(false);

  useEffect(() => {
    localStorage.setItem('teamName', name);
  }, [name]);

  const handleDoubleClick = () => {
    setIsEditing(true);
  };

  const handleBlur = () => {
    setIsEditing(false);
  };

  const handleChange = (e: React.ChangeEvent<HTMLInputElement>) => {
    let newName = e.target.value;

    if (newName.length > 20) {
      newName = newName.substring(0, 20);
    }

    setName(newName);
  };

  return (
    <div className="team-wrapper">
      <header>
        {isEditing ? (
          <input
            type="text"
            value={name}
            onChange={handleChange}
            onBlur={handleBlur}
            autoFocus
            className="input-team-name"
          />
        ) : (
          <h2 className="team-name" onDoubleClick={handleDoubleClick}>
            {name}
          </h2>
        )}
        <div>
          <button className="edit-name-button" onClick={() => setIsEditing(!isEditing)}>
            {isEditing ? (
              <IoIosSave size={30} className="edit-name-icon" />
            ) : (
              <RiPencilFill size={30} className="edit-name-icon" />
            )}
          </button>
        </div>
      </header>

      <section className="team-section">
        {pokemons.length > 0 ? (
          pokemons.map((pokemon) => (
            <div className="pokemon-card-container" key={pokemon.pokedexNumber}>
              <PokemonCard
                pokemon={pokemon}
                isFavorite={isFavorited(pokemon)}
                onToggleFavorite={() => onToggleFavorite(pokemon)}
              />
            </div>
          ))
        ) : (
          <p className="no-pokemons-text">No Pok√©mons in your team.</p>
        )}
      </section>
    </div>
  );
}
