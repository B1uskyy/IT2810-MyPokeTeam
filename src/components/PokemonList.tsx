import { ListPokemon } from '../interfaces/pokemons.ts';
import PokemonCard from './PokemonCard.tsx';
import '../styles/pokemonList.css';

interface PokemonListProps {
  pokemons: ListPokemon[];
  onToggleFavorite: (pokemon: ListPokemon) => void;
  isFavorited: (pokemon: ListPokemon) => boolean;
}

const PokemonList = ({ pokemons, onToggleFavorite, isFavorited }: PokemonListProps) => {
  return (
    <div className="pokemon-grid">
      {pokemons.length > 0 ? (
        pokemons.map((pokemon) => (
          <PokemonCard
            key={pokemon.name}
            pokemon={pokemon}
            isFavorite={isFavorited(pokemon)}
            onToggleFavorite={() => onToggleFavorite(pokemon)}
          />
        ))
      ) : (
        <div>No Pok√©mon available</div>
      )}
    </div>
  );
};

export default PokemonList;
