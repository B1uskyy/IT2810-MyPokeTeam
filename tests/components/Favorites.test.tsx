import { render, screen } from '@testing-library/react';
import { Favorites } from '../../src/components/Favorites';
import { MemoryRouter, Route, Routes } from 'react-router-dom';
import { vi } from 'vitest';

vi.mock('../../src/components/PokemonCard', () => ({
  // used gpt here:"The pokemoncard components all have different rendered names: <code> how can I check that 6 children are rendered?"
  default: () => <div>PokemonCard</div>,
}));

const testData = {
  name: 'Chris',
};

describe('Favorites component', () => {
  it('renders with correct props ', () => {
    render(
      <MemoryRouter initialEntries={['/team']}>
        <Routes>
          <Route path="/team" element={<Favorites name={testData.name} />} />
        </Routes>
      </MemoryRouter>,
    );
    expect(screen.getByText('Team Chris')).toBeInTheDocument();
  });

  it('renders 6 PokemonCard components', () => {
    render(
      <MemoryRouter initialEntries={['/team']}>
        <Routes>
          <Route path="/team" element={<Favorites name={testData.name} />} />
        </Routes>
      </MemoryRouter>,
    );
    const pokemonCards = screen.getAllByText('PokemonCard');
    expect(pokemonCards).toHaveLength(6);
  });
});
