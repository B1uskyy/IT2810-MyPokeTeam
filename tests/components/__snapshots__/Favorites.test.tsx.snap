// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Favorites component > matches snapshot with correct props 1`] = `
<DocumentFragment>
  <div
    class="team-wrapper"
  >
    <header>
      <h2
        class="team-name"
      >
        Team name
      </h2>
      <div>
        <button
          class="edit-name-button"
        >
          <svg
            class="edit-name-icon"
            fill="currentColor"
            height="30"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 24 24"
            width="30"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M12.8995 6.85453L17.1421 11.0972L7.24264 20.9967H3V16.754L12.8995 6.85453ZM14.3137 5.44032L16.435 3.319C16.8256 2.92848 17.4587 2.92848 17.8492 3.319L20.6777 6.14743C21.0682 6.53795 21.0682 7.17112 20.6777 7.56164L18.5563 9.68296L14.3137 5.44032Z"
            />
          </svg>
        </button>
      </div>
    </header>
    <section
      class="team-section"
    >
      <div
        class="pokemon-card-container"
      >
        <div>
          <p>
            Bulbasaur
          </p>
          <button
            data-testid="favorite-button-1"
          >
            Unfavorite
          </button>
        </div>
      </div>
      <div
        class="pokemon-card-container"
      >
        <div>
          <p>
            Ivysaur
          </p>
          <button
            data-testid="favorite-button-2"
          >
            Favorite
          </button>
        </div>
      </div>
    </section>
  </div>
</DocumentFragment>
`;
